### PR TITLE
[master] Upgrade Rust toolchain

### PR DESCRIPTION
**Pull request by onalante-msft**
Unifies toolchain files.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [x] local testing done.

### CODE REVIEW

File: scripts/linux/generic-rust/install.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ function install_rust()
     # but it's simpler to just always `update` whatever toolchain it is. `update` installs the toolchain
     # if it hasn't already been installed, so this also works for pinned versions.
     source $HOME/.cargo/env
-    rustup update "$(< "$PROJECT_ROOT/rust-toolchain")"
+    ( cd $PROJECT_ROOT; rustup show active-toolchain )
```
Comment (vadim-kovalyov) [inline scripts/linux/generic-rust/install.sh:52]: not sure why this change?

**Review by vadim-kovalyov — APPROVED** [review]
Thanks for doing this @onalante-msft!

File: scripts/linux/generic-rust/install.sh (hunk 1 above)
Comment (onalante-msft) [reply scripts/linux/generic-rust/install.sh:52]: This is because I changed the toolchain override file to TOML, meaning its contents cannot be used directly as a command-line argument like before. As long as build invocations are run from each project root, this command should actually be unnecessary.

File: mqtt/rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1 @@
+../rust-toolchain.toml
```
Comment (arsing) [inline mqtt/rust-toolchain.toml:1]: Are the symlinks needed? Does rustup not look for it above the workspace root?

File: scripts/linux/generic-rust/install.sh (hunk 1 above)
Comment (arsing) [reply scripts/linux/generic-rust/install.sh:52]: The file moved one level up, and it's true that there's no need to keep the path in sync when `rustup` can look it up for us.

**Review by arsing — CHANGES_REQUESTED** [review]

File: mqtt/rust-toolchain.toml (hunk 2 above)
Comment (onalante-msft) [reply mqtt/rust-toolchain.toml:1]: It seems like rustup does look for it in the workspace root. I will remove the symlinks.

**Review by arsing — APPROVED** [review]

File: scripts/linux/generic-rust/install.sh [hunk 3]
```diff
@@ -49,7 +49,7 @@ function install_rust()
     # but it's simpler to just always `update` whatever toolchain it is. `update` installs the toolchain
     # if it hasn't already been installed, so this also works for pinned versions.
     source $HOME/.cargo/env
-    rustup update "$(< "$PROJECT_ROOT/rust-toolchain")"
+    rustup show active-toolchain
```
Comment (arsing) [inline scripts/linux/generic-rust/install.sh:52]: Shouldn't this be `rustup update "$(rustup show active-toolchain)"` ?

File: scripts/linux/generic-rust/install.sh (hunk 1 above)
Comment (onalante-msft) [reply scripts/linux/generic-rust/install.sh]: `rustup show active-toolchain` coerces `rustup` into installing the toolchain specified in `rust-toolchain.toml` during the current build step rather than later on during a `cargo` invocation.

File: scripts/linux/generic-rust/install.sh (hunk 3 above)
Comment (onalante-msft) [reply scripts/linux/generic-rust/install.sh:52]: As far as I can tell, `rustup show active-toolchain` always synchronizes the tracked channel and updates if necessary.

Comment (arsing) [reply scripts/linux/generic-rust/install.sh:52]: I just tried it on my machine where the default is nightly.

```sh
$ rustup show active-toolchain

nightly-x86_64-unknown-linux-gnu (default)

$ rustup update nightly

info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
716.3 KiB / 716.3 KiB (100 %) 703.5 KiB/s in  1s ETA:  0s
info: latest update on 2022-02-14, rust version 1.60.0-nightly (1e12aef3f 2022-02-13)
info: downloading component 'clippy'
...
```

So no, I don't think it implicitly updates.

Comment (onalante-msft) [reply scripts/linux/generic-rust/install.sh:52]: Interesting, will fix.

Comment (arsing) [reply scripts/linux/generic-rust/install.sh:52]: Also, you can't directly do what I suggested, because

```sh
$ rustup update "$(rustup show active-toolchain)"

info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu (default)'
info: latest update on 2022-02-14, rust version 1.60.0-nightly (1e12aef3f 2022-02-13)
error: target 'x86_64-unknown-linux-gnu (default)' not found in channel.  Perhaps check https://doc.rust-lang.org/nightly/rustc/platform-support.html for available targets
```

So you'd need to `sed` out the trailing ` (default)`.

(You could do just `rustup update` but the disadvantage is that it updates all toolchains, not just the installed one. But maybe that's not a big deal.)

Comment (onalante-msft) [reply scripts/linux/generic-rust/install.sh]: Adding a `sed` is a small concession compared to downloading a completely new toolchain, so I will go with the `sed` route as you suggested.

**Review by arsing — APPROVED** [review]
